File: scripts/ME_Analysis/MW_Analysis/NN_AN/selectedNNs/MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500.py
from math import exp

from math import tanh

class MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500:
	def value(self,index,in0,in1,in2):
		self.input0 = (in0 - 22.1454)/2.02206
		self.input1 = (in1 - 25.1712)/2.09579
		self.input2 = (in2 - 13.8698)/2.37204
		if index==0: return self.neuron0x5a24cf0();
		return 0.
	def neuron0x5a10900(self):
		return self.input0
	def neuron0x5a10c40(self):
		return self.input1
	def neuron0x5a21b80(self):
		return self.input2
	def neuron0x5a21f60(self):
		input = -0.0588938
		input = input + self.synapse0x59f75c0()
		input = input + self.synapse0x59d9080()
		input = input + self.synapse0x5a22210()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a22250(self):
		input = 2.79388
		input = input + self.synapse0x5a22590()
		input = input + self.synapse0x5a225d0()
		input = input + self.synapse0x5a22610()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a22650(self):
		input = -1.46534
		input = input + self.synapse0x5a22990()
		input = input + self.synapse0x5a229d0()
		input = input + self.synapse0x5a22a10()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a22a50(self):
		input = 0.521133
		input = input + self.synapse0x5a22d90()
		input = input + self.synapse0x5a22dd0()
		input = input + self.synapse0x5a22e10()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a22e50(self):
		input = -0.310288
		input = input + self.synapse0x5a23190()
		input = input + self.synapse0x5a231d0()
		input = input + self.synapse0x5a23210()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a23250(self):
		input = -0.704482
		input = input + self.synapse0x5a23590()
		input = input + self.synapse0x5a235d0()
		input = input + self.synapse0x59d8bb0()
		input = input + self.synapse0x59d8bf0()
		input = input + self.synapse0x5a23720()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a23760(self):
		input = 0.240326
		input = input + self.synapse0x5a23aa0()
		input = input + self.synapse0x5a23ae0()
		input = input + self.synapse0x5a23b20()
		input = input + self.synapse0x5a23b60()
		input = input + self.synapse0x5a23ba0()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a23be0(self):
		input = -0.729438
		input = input + self.synapse0x5a23f20()
		input = input + self.synapse0x5a23f60()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a23fa0(self):
		input = 1.18112
		input = input + self.synapse0x5a242e0()
		input = input + self.synapse0x5a24320()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a24360(self):
		input = -1.11026
		input = input + self.synapse0x5a246a0()
		input = input + self.synapse0x5a246e0()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a24720(self):
		input = 0.850224
		input = input + self.synapse0x5a24a60()
		input = input + self.synapse0x5a24aa0()
		input = input + self.synapse0x59f7540()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x5a24cf0(self):
		input = 6.06658
		input = input + self.synapse0x59f7580()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def synapse0x59f75c0(self):
		return (self.neuron0x5a10900()*-0.919716)
	def synapse0x59d9080(self):
		return (self.neuron0x5a10c40()*0.453358)
	def synapse0x5a22210(self):
		return (self.neuron0x5a21b80()*3.03902)
	def synapse0x5a22590(self):
		return (self.neuron0x5a10900()*0.820969)
	def synapse0x5a225d0(self):
		return (self.neuron0x5a10c40()*-2.13804)
	def synapse0x5a22610(self):
		return (self.neuron0x5a21b80()*1.19974)
	def synapse0x5a22990(self):
		return (self.neuron0x5a10900()*0.653192)
	def synapse0x5a229d0(self):
		return (self.neuron0x5a10c40()*2.00153)
	def synapse0x5a22a10(self):
		return (self.neuron0x5a21b80()*-3.6891)
	def synapse0x5a22d90(self):
		return (self.neuron0x5a10900()*-1.11734)
	def synapse0x5a22dd0(self):
		return (self.neuron0x5a10c40()*1.93581)
	def synapse0x5a22e10(self):
		return (self.neuron0x5a21b80()*-2.11766)
	def synapse0x5a23190(self):
		return (self.neuron0x5a10900()*0.484215)
	def synapse0x5a231d0(self):
		return (self.neuron0x5a10c40()*-1.28345)
	def synapse0x5a23210(self):
		return (self.neuron0x5a21b80()*2.10527)
	def synapse0x5a23590(self):
		return (self.neuron0x5a21f60()*1.0904)
	def synapse0x5a235d0(self):
		return (self.neuron0x5a22250()*2.85411)
	def synapse0x59d8bb0(self):
		return (self.neuron0x5a22650()*0.641853)
	def synapse0x59d8bf0(self):
		return (self.neuron0x5a22a50()*-2.27269)
	def synapse0x5a23720(self):
		return (self.neuron0x5a22e50()*-3.01609)
	def synapse0x5a23aa0(self):
		return (self.neuron0x5a21f60()*0.5849)
	def synapse0x5a23ae0(self):
		return (self.neuron0x5a22250()*-0.644956)
	def synapse0x5a23b20(self):
		return (self.neuron0x5a22650()*-0.732079)
	def synapse0x5a23b60(self):
		return (self.neuron0x5a22a50()*0.925675)
	def synapse0x5a23ba0(self):
		return (self.neuron0x5a22e50()*1.3573)
	def synapse0x5a23f20(self):
		return (self.neuron0x5a23250()*2.76996)
	def synapse0x5a23f60(self):
		return (self.neuron0x5a23760()*-0.771015)
	def synapse0x5a242e0(self):
		return (self.neuron0x5a23250()*-3.91347)
	def synapse0x5a24320(self):
		return (self.neuron0x5a23760()*1.16068)
	def synapse0x5a246a0(self):
		return (self.neuron0x5a23250()*4.01207)
	def synapse0x5a246e0(self):
		return (self.neuron0x5a23760()*-1.71203)
	def synapse0x5a24a60(self):
		return (self.neuron0x5a23be0()*-3.03156)
	def synapse0x5a24aa0(self):
		return (self.neuron0x5a23fa0()*5.16055)
	def synapse0x59f7540(self):
		return (self.neuron0x5a24360()*-5.26779)
	def synapse0x59f7580(self):
		return (self.neuron0x5a24720()*-12.482)

File: test/ME_Analysis/Merging/MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500.cxx
#include "MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500.h"
#include <cmath>

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::Value(int index,double in0,double in1,double in2) {
   input0 = (in0 - 22.1454)/2.02206;
   input1 = (in1 - 25.1712)/2.09579;
   input2 = (in2 - 13.8698)/2.37204;
   switch(index) {
     case 0:
         return neuron0x5a24cf0();
     default:
         return 0.;
   }
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::Value(int index, double* input) {
   input0 = (input[0] - 22.1454)/2.02206;
   input1 = (input[1] - 25.1712)/2.09579;
   input2 = (input[2] - 13.8698)/2.37204;
   switch(index) {
     case 0:
         return neuron0x5a24cf0();
     default:
         return 0.;
   }
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a10900() {
   return input0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a10c40() {
   return input1;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a21b80() {
   return input2;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a21f60() {
   double input = -0.0588938;
   input += synapse0x59f75c0();
   input += synapse0x59d9080();
   input += synapse0x5a22210();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a21f60() {
   double input = input0x5a21f60();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a22250() {
   double input = 2.79388;
   input += synapse0x5a22590();
   input += synapse0x5a225d0();
   input += synapse0x5a22610();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a22250() {
   double input = input0x5a22250();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a22650() {
   double input = -1.46534;
   input += synapse0x5a22990();
   input += synapse0x5a229d0();
   input += synapse0x5a22a10();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a22650() {
   double input = input0x5a22650();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a22a50() {
   double input = 0.521133;
   input += synapse0x5a22d90();
   input += synapse0x5a22dd0();
   input += synapse0x5a22e10();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a22a50() {
   double input = input0x5a22a50();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a22e50() {
   double input = -0.310288;
   input += synapse0x5a23190();
   input += synapse0x5a231d0();
   input += synapse0x5a23210();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a22e50() {
   double input = input0x5a22e50();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a23250() {
   double input = -0.704482;
   input += synapse0x5a23590();
   input += synapse0x5a235d0();
   input += synapse0x59d8bb0();
   input += synapse0x59d8bf0();
   input += synapse0x5a23720();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a23250() {
   double input = input0x5a23250();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a23760() {
   double input = 0.240326;
   input += synapse0x5a23aa0();
   input += synapse0x5a23ae0();
   input += synapse0x5a23b20();
   input += synapse0x5a23b60();
   input += synapse0x5a23ba0();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a23760() {
   double input = input0x5a23760();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a23be0() {
   double input = -0.729438;
   input += synapse0x5a23f20();
   input += synapse0x5a23f60();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a23be0() {
   double input = input0x5a23be0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a23fa0() {
   double input = 1.18112;
   input += synapse0x5a242e0();
   input += synapse0x5a24320();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a23fa0() {
   double input = input0x5a23fa0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a24360() {
   double input = -1.11026;
   input += synapse0x5a246a0();
   input += synapse0x5a246e0();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a24360() {
   double input = input0x5a24360();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a24720() {
   double input = 0.850224;
   input += synapse0x5a24a60();
   input += synapse0x5a24aa0();
   input += synapse0x59f7540();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a24720() {
   double input = input0x5a24720();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::input0x5a24cf0() {
   double input = 6.06658;
   input += synapse0x59f7580();
   return input;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::neuron0x5a24cf0() {
   double input = input0x5a24cf0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59f75c0() {
   return (neuron0x5a10900()*-0.919716);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59d9080() {
   return (neuron0x5a10c40()*0.453358);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22210() {
   return (neuron0x5a21b80()*3.03902);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22590() {
   return (neuron0x5a10900()*0.820969);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a225d0() {
   return (neuron0x5a10c40()*-2.13804);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22610() {
   return (neuron0x5a21b80()*1.19974);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22990() {
   return (neuron0x5a10900()*0.653192);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a229d0() {
   return (neuron0x5a10c40()*2.00153);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22a10() {
   return (neuron0x5a21b80()*-3.6891);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22d90() {
   return (neuron0x5a10900()*-1.11734);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22dd0() {
   return (neuron0x5a10c40()*1.93581);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a22e10() {
   return (neuron0x5a21b80()*-2.11766);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23190() {
   return (neuron0x5a10900()*0.484215);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a231d0() {
   return (neuron0x5a10c40()*-1.28345);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23210() {
   return (neuron0x5a21b80()*2.10527);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23590() {
   return (neuron0x5a21f60()*1.0904);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a235d0() {
   return (neuron0x5a22250()*2.85411);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59d8bb0() {
   return (neuron0x5a22650()*0.641853);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59d8bf0() {
   return (neuron0x5a22a50()*-2.27269);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23720() {
   return (neuron0x5a22e50()*-3.01609);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23aa0() {
   return (neuron0x5a21f60()*0.5849);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23ae0() {
   return (neuron0x5a22250()*-0.644956);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23b20() {
   return (neuron0x5a22650()*-0.732079);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23b60() {
   return (neuron0x5a22a50()*0.925675);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23ba0() {
   return (neuron0x5a22e50()*1.3573);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23f20() {
   return (neuron0x5a23250()*2.76996);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a23f60() {
   return (neuron0x5a23760()*-0.771015);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a242e0() {
   return (neuron0x5a23250()*-3.91347);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a24320() {
   return (neuron0x5a23760()*1.16068);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a246a0() {
   return (neuron0x5a23250()*4.01207);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a246e0() {
   return (neuron0x5a23760()*-1.71203);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a24a60() {
   return (neuron0x5a23be0()*-3.03156);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x5a24aa0() {
   return (neuron0x5a23fa0()*5.16055);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59f7540() {
   return (neuron0x5a24360()*-5.26779);
}

double MLP_Higgs_vs_TT_MM_N_CSV_2012_comb_ZH125_5_2_3_1_500::synapse0x59f7580() {
   return (neuron0x5a24720()*-12.482);
}

